model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/remote/RemoteUserSessionProvider.java

```diff
@@ -462,9 +461,10 @@ private Stream<UserSessionModel> streamUserSessionByClientId(RealmModel realm, S
         var userSessionTx = getUserSessionTransaction(offline);
         return Flowable.fromIterable(QueryHelper.toCollection(userSessionIdQuery, QueryHelper.SINGLE_PROJECTION_TO_STRING))
                 .flatMapMaybe(userSessionTx::maybeGet, false, MAX_CONCURRENT_REQUESTS)
-                .concatMapMaybe(session -> maybeInitUserSessionFromQuery(session, realm, offline))
+                .blockingStream()
```

Comment:
missing the batch size
```suggestion
                .blockingStream(batchSize)
```